Added example template for Google Deployment Manager feature of CAM (#2554)
CenturyLinkCloud/PublicKB: Cloud Application Manager/Automating Deployments/google-deployment-manager.md

- [Overview](#overview)
- [Audience](#audience)
- [Prerequisites](#prerequisites)

> - [Getting started with Google Deployment Templates](#getting-started-with-google-deployment-templates)

Reviewer: Please, leave it as it was (it's best for search engine scoring to include the keywords several times)

- [Google Deployment Templates](#google-deployment-templates)

> - [Example Template](#example-template)

Reviewer: ```suggestion
- [Sample Template](#sample-template)
```

### Prerequisites

- An active Cloud Application Manager account
- An existing [Google Compute Provider](../Deploying Anywhere/using-google-cloud.md) configured in Cloud Application Manager

> ### Getting started with Google Deployment Templates

Reviewer: Leave it as it was

might get blocked, because it can not clean up a resource it has deployed,
if a different instance has already deleted it.

A good practice to avoid these scenarios is to prefix the name of each resource with the **name of the deployment**,
which happens to be the *service-id* in Cloud Application Manager.

> ```Jinja

Reviewer: Would it be `yaml` best?

Author: Depends on the rendered style, but really nothing spectacular to see in Yaml, but with Jinja it highlights the variable substitutions, which is informative.

[Google's Documentation](https://cloud.google.com/deployment-manager/docs/configuration/expose-information-outputs).
Any Outputs defined in templates are going to end up as output variables in Cloud Application Manager. 

> ```Jinja

Reviewer: Would it be `yaml` best?

![Resources deployed with Google Deployment Manager](../../images/cloud-application-manager/google-deployment-manager/resources.png)

> ### Example Template

Reviewer: ```suggestion
### Sample Template
```

The following template deploys a single virtual machine instance with a firewall rule.

To use this template in a Template Box, you need to define a mandatory variable named `machineType` with value of a valid
Google Compute Engine virtual [machine type](https://cloud.google.com/compute/docs/machine-types).
Here are some possible values to define it as an Option type variable: `f1-micro,g1-small,n1-standard-1`

**Important to note** that in the following template, the `name` property of each resource gets a derived value,
`{{env['deployment']}}` as a prefix, which contains the deployment-ID, and therefore gets a new distinct value
each time a new Instance is deployed. If we used static names for the resources, a second deployed Instance
would update the already existing resource, instead of creating a new one, and would delete the only resource
when terminated, leaving the remaining Instance referring to a non-existing resource, and therefore fail to

> terminate in CAM. And mind the dire consequences of unintentionally deleting resources.

Reviewer: ```suggestion
terminate in CAM. And also bear in mind the direct consequences of unintentionally deleting resources.
```

Author: Now it sounds:
`And also bear in mind the dire threat of unintentionally deleting resources.`

> ```Jinja

Reviewer: ```suggestion
```yaml
```

    disks:
    - deviceName: boot
      type: PERSISTENT
      boot: true
      autoDelete: true
      initializeParams:
        sourceImage: projects/debian-cloud/global/images/family/debian-9
    networkInterfaces:
    - network: global/networks/default
      accessConfigs:
      - name: External NAT
        type: ONE_TO_ONE_NAT
    metadata:
      items:
      - key: startup-script
        value: |
          #!/bin/bash -e
          apt update && apt -y install nginx-light
    tags:
      items:
      - {{ name }}-tcp-80
- name: {{ env['deployment'] }}-tcp-80
  type: compute.v1.firewall
  properties:
    allowed:
    - IPProtocol: TCP
      ports:
      - '80'
    network: global/networks/default
    sourceRanges:
    - 0.0.0.0/0
    targetTags:
    - {{name}}-tcp-80

outputs:
- name: internalIP
  value: $(ref.{{ env['deployment'] }}-vm.networkInterfaces[0].networkIP)
- name: ip
  value: $(ref.{{ env['deployment'] }}-vm.networkInterfaces[0].accessConfigs[0].natIP)
- name: port
  value: 80
```

This template defines `internalIP`, `ip` and `port` output variables. You can check their output value in the Lifecycle Editor
of the deployed Instance.

> **To deploy this template:**

Reviewer: ```suggestion
#### To deploy this template
```
